script/metric/extract_conflict_mineral_report.rb

```diff
@@ -0,0 +1,116 @@
+require File.expand_path('../../../config/environment',  __FILE__)
+require 'colorize'
+require 'json'
+require 'byebug'
+require 'rubygems'
+require 'csv'
+
+file_name = 'script/metric/data/certh.json'
+
+text = File.open(file_name).read
+
+def generate_aliases_hash
+  aliases_hash = {}
+  aliases_cards = Card.search right: 'aliases',
+                              left: { type_id: Card::WikirateCompanyID }
+  aliases_cards.each do |aliases_card|
+    aliases_card.item_names.each do |name|
+      aliases_hash[name.downcase] = aliases_card.cardname.left
+    end
+  end
+  aliases_hash
+end
+
+def potential_company name
+  result = Card.search type: 'company', name: ['match', name]
+  return nil if result.empty?
+  result
+end
+
+def company_skip_list
+  %w(BROO Off ACE SAS WPS)
+end
+
+def correct_company_name company, aliases, aliases_hash, skip_list
+  return company if Card.exists?(company)
+  aliases.each do |al|
+    if (found_company = aliases_hash[al.downcase])
+      return found_company
+    end
+  end
+  return company if skip_list.include?(company)
+  if (potential_com = potential_company(company))
+    return potential_com.map(&:name)[0]
+  end
+  company
+end
+
+begin
+
```

Comment:
no more stream-of-consciousness coding, please.  all this stuff needs to be in methods.  The actual extraction should be one or two method calls, and each method needs to be tight and well named.